### PR TITLE
Add support form blatdb

### PR DESCRIPTION
**Pull request by mboudet**
- Moved the file copying to the python script and added safeguards
- Convert .fasta to .2bit, and add the path to it to the organism in Apollo

### CODE REVIEW

File: tools/apollo/create_or_update_organism.py [hunk 1]
```diff
@@ -72,7 +108,8 @@
             args.jbrowse,
             genus=args.genus,
             species=args.species,
-            public=args.public
+            public=args.public,
+	        blatdb = blat_db
```
Comment (abretaud) [inline tools/apollo/create_or_update_organism.py]: tab/space problem here?

**Comment by abretaud** [comment]
Related things:
- we have an up-to-date Apollo image at https://github.com/abretaud/docker-apollo/tree/bipaa, we should push most of the changes to https://github.com/galaxy-genome-annotation/docker-apollo
- for this blat thing to work, we also need to add the blat binary in the image
- and enable it probably as https://genomearchitect.readthedocs.io/en/latest/Configure.html#main-configuration

**Comment by erasche** [comment]
> for this blat thing to work, we also need to add the blat binary in the image

The license of this won't allow commercial use, that's why it has never been included before :(

**Comment by abretaud** [comment]
Aaaaah damned, I forgot about that ;(

**Comment by abretaud** [comment]
Ok, so can we say that we do not ship blat in the docker image, but we generate the 2bit index anyway (or maybe make it optional?) and we can choose to mount the blat binary when running the container (and enabling the blat tool in the apollo config)?
Here at bipaa we would use it as we know there is no commercial use (at least for many of our genomes)

**Comment by erasche** [comment]
Sounds great @abretaud!

**Comment by mboudet** [comment]
It might be good to check for an env variable (like BLAT_ENABLED) if it's not going to be always installed.
If only to save the time spend converting the fasta file.

File: tools/apollo/create_or_update_organism.py [hunk 2]
```diff
@@ -56,7 +101,8 @@
             # mandatory
             genus=args.genus,
             species=args.species,
-            public=args.public
+            public=args.public,
+            blatdb=blat_db
```
Comment (abretaud) [inline tools/apollo/create_or_update_organism.py]: we should send the blatdb path only if the 2bit index was generated

File: tools/apollo/create_or_update_organism.py [hunk 3]
```diff
@@ -72,7 +118,8 @@
             args.jbrowse,
             genus=args.genus,
             species=args.species,
-            public=args.public
+            public=args.public,
+            blatdb=blat_db
```
Comment (abretaud) [inline tools/apollo/create_or_update_organism.py]: here too

File: tools/apollo/create_or_update_organism.py [hunk 4]
```diff
@@ -24,8 +39,38 @@
...
+    # Copy files
+    shutil.copytree(args.jbrowse_old, args.jbrowse)
+
+    path_fasta = args.jbrowse + '/seq/genome.fasta'
+    path_2bit = args.jbrowse + '/seq/genome.2bit'
+
+    # Convert fasta if existing
+    if(IsBlatEnabled() and os.path.exists(path_fasta)):
+        arg = ['faToTwoBit', path_fasta, path_2bit]
+        tmp_stderr = tempfile.NamedTemporaryFile(prefix="tmp-data-manager-twobit-builder-stderr")
```
Comment (abretaud) [inline tools/apollo/create_or_update_organism.py]: remove reference to data-manager

File: tools/apollo/create_or_update_organism.py [hunk 5]
```diff
@@ -4,18 +4,33 @@
...
+        return True
+    else:
+        return False
+
+
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Create or update an organism in an Apollo instance')
     WAAuth(parser)
-
+    parser.add_argument('jbrowse_old', help='Old JBrowse Data Directory')
```
Comment (abretaud) [inline tools/apollo/create_or_update_organism.py]: rename it to jbrowse_src maybe?

File: tools/apollo/create_or_update_organism.py (hunk 2 above)
Comment (mboudet) [reply tools/apollo/create_or_update_organism.py]: blat_db is set to none if the 2bit was not created, wgich is the default value in both update and create function

Comment (abretaud) [reply tools/apollo/create_or_update_organism.py]: ah ok nevermind

**Pull request merged**